src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py

```diff
@@ -344,8 +344,11 @@ def __call__(
         t_start = max(num_inference_steps - init_timestep + offset, 0)
 
         # Some schedulers like PNDM have timesteps as arrays
-        # It's more optimzed to move all timesteps to correct device beforehand
-        timesteps_tensor = torch.tensor(self.scheduler.timesteps[t_start:], device=self.device)
```

Comment:
I think:

```
 timesteps_tensor = torch.tensor(self.scheduler.timesteps[t_start:], device=self.device)
```

was the better API - i.e. I think we should try to get rid of the `if else` statement by forcing everything to be in PyTorch

Reply:
Could we try to not use the if-else statement here?